### PR TITLE
CTL9-18-Connect API with Cocktails UI

### PR DESCRIPTION
**Pull request by gordanailic**

### CODE REVIEW

File: app/src/main/java/com/example/cocktails/modules/AppModule.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    ): CocktailsService {
+        return Retrofit.Builder()
+            .baseUrl("https://www.thecocktaildb.com/api/json/v1/1/")
+            .addConverterFactory(GsonConverterFactory.create(gson))
+            .addCallAdapterFactory(CoroutineCallAdapterFactory())
+            .client(httpClient)
+            .build()
+            .create(CocktailsService::class.java)
+    }
+    @Provides
```
Comment (Levi9Daniel) [inline app/src/main/java/com/example/cocktails/modules/AppModule.kt]: Remove unnecessary provided objects

File: app/src/main/java/com/example/cocktails/modules/AppModule.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Provides
+    @Singleton
+    fun provideMyApi(
+        gson: Gson,
+        httpClient: OkHttpClient
+    ): CocktailsService {
+        return Retrofit.Builder()
+            .baseUrl("https://www.thecocktaildb.com/api/json/v1/1/")
```
Comment (Levi9Daniel) [inline app/src/main/java/com/example/cocktails/modules/AppModule.kt]: Extract to a constant

File: app/src/main/java/com/example/cocktails/modules/AppModule.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            .addConverterFactory(GsonConverterFactory.create(gson))
+            .addCallAdapterFactory(CoroutineCallAdapterFactory())
```
Comment (Levi9Daniel) [inline app/src/main/java/com/example/cocktails/modules/AppModule.kt]: We should remove this

File: app/src/main/java/com/example/cocktails/presentation/viewmodel/CocktailViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.cocktails.data.models.Cocktail
+import com.example.cocktails.data.models.Resource
+import com.example.cocktails.data.repositories.CocktailRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.delay
```
Comment (Levi9Daniel) [inline app/src/main/java/com/example/cocktails/presentation/viewmodel/CocktailViewModel.kt]: Optimize imports

File: app/src/main/java/com/example/cocktails/presentation/viewmodel/CocktailViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+@HiltViewModel
+class CocktailViewModel @Inject constructor(
+    private val cocktailRepository: CocktailRepository
+) : ViewModel() {
+    private val _cocktails = MutableLiveData<Resource<List<Cocktail>>>()
+    val cocktails: LiveData<Resource<List<Cocktail>>> get() = _cocktails
+
+    private val handler = CoroutineExceptionHandler { _, exception ->
+        _cocktails.value = Resource.Error("Failed to fetch cocktails ${exception.message}")
```
Comment (Levi9Daniel) [inline app/src/main/java/com/example/cocktails/presentation/viewmodel/CocktailViewModel.kt]: All strings should be extracted to strings.xml file. Resources should be injected via Hilt. Do this for all the strings/errors

File: app/src/main/java/com/example/cocktails/presentation/view/fragments/cocktails/recycler/diff/CoctailDiffcallback.kt [hunk 6]
```diff
@@ -12,6 +12,7 @@ class CoctailDiffcallback : DiffUtil.ItemCallback<Cocktail>() {
     override fun areContentsTheSame(oldItem: Cocktail, newItem: Cocktail): Boolean {
         return oldItem.name == newItem.name &&
                 oldItem.image == newItem.image &&
+                oldItem.alcoholic == newItem.alcoholic &&
                 oldItem.favorite == newItem.favorite
```
Comment (Levi9Daniel) [inline app/src/main/java/com/example/cocktails/presentation/view/fragments/cocktails/recycler/diff/CoctailDiffcallback.kt:16]: Rename this file to CocktailDiffcallback

**Review by Levi9Daniel — DISMISSED** [review]

**Review by a-radovanovic-levi9-com — APPROVED** [review]

**Pull request merged**